src/ReadTrainer/ReadTrainer.js

```diff
@@ -19,14 +23,36 @@ const PlayLoop = inject("store")(
       text: ""
     };
 
-    pickText = () => {
+    pickText = (previousText = "") => {
       const store = this.props.store.readTrainer;
+      const trimmedDictionary = trimDictionary(
```

Comment:
Is it necessary to do the trim every time, or could we get away with just when the settings change? trimDictionary() iterates over the full dict twice and sorts it once, so it's not exactly cheap...

Reply:
You're right, this could be premature optimization. I was indeed thinking of slower devices but maybe it's fine as is. Would you mind just adding a comment about it for now?

Reply:
Sure thing.